### PR TITLE
fixes #10414 - Service level doesn't copy when cloning activation key

### PR DESCRIPTION
**Pull request by cfouant**

### CODE REVIEW

File: app/controllers/katello/api/v2/activation_keys_controller.rb [hunk 1]
```diff
@@ -87,16 +87,16 @@ def copy
...
+                  :release_version => @activation_key.release_version,
+                  :auto_attach     => @activation_key.auto_attach
+               )
       @activation_key.content_overrides.each do |content|
         @new_activation_key.set_content_override(content['contentLabel'], content[:name], content[:value])
       end
-      @activation_key.get_key_pools.each do |pool|
-        @new_activation_key.subscribe(pool[:id], pool[:amount])
+      @activation_key.subscriptions.each do |subscription|
+        @new_activation_key.subscribe(subscription[:id])
```
Comment (ehelms) [inline app/controllers/katello/api/v2/activation_keys_controller.rb]: Does this make calls to Candlepin? If so, we ought to open an issue to dynflowize this.

Comment (cfouant) [inline app/controllers/katello/api/v2/activation_keys_controller.rb]: It does, and I have a redmine ticket in to dynflowize the copy function. http://projects.theforeman.org/issues/9432 
Just wanted to get to this first.

Comment (ehelms) [inline app/controllers/katello/api/v2/activation_keys_controller.rb]: Ahh, great!

File: app/controllers/katello/api/v2/activation_keys_controller.rb [hunk 2]
```diff
@@ -306,7 +305,6 @@ def activation_key_params
         key_params[:unlimited_content_hosts] = false if max_hosts
         key_params[:max_content_hosts] = nil if unlimited
       end
-
```
Comment (ehelms) [inline app/controllers/katello/api/v2/activation_keys_controller.rb]: Are these blank line insertions needed for this change?

Comment (cfouant) [inline app/controllers/katello/api/v2/activation_keys_controller.rb]: I was removing them, but I can put them back if you like

Comment (ehelms) [inline app/controllers/katello/api/v2/activation_keys_controller.rb]: If they aren't relative or required for the change, I prefer they not be included.

Comment (cfouant) [inline app/controllers/katello/api/v2/activation_keys_controller.rb]: @ehelms - changes reverted!

**Comment by ehelms** [comment]
Thoughts on a test here to prevent a future regression?

**Comment by ehelms** [comment]
ping @cfouant

**Comment by cfouant** [comment]
@ehelms - tests finished. Thanks for the guidance with the stubbing!

**Comment by jlsherrill** [comment]
ACK

**Comment by cfouant** [comment]
@jlsherrill :+1: :100: THANK YOU FOR YOUR REVIEW!!!

**Pull request merged**